sklearn/metrics/cluster/unsupervised.py

```diff
@@ -56,6 +65,18 @@ def silhouette_score(X, labels, metric='euclidean', sample_size=None,
         <sklearn.metrics.pairwise.pairwise_distances>`. If X is the distance
         array itself, use ``metric="precomputed"``.
 
+    block_size : int, optional, default=64
```

Comment:
I'm not committed to it. (Nor am I, by the way, committed to this being per job. I'm happy for it to be divided by the jobs.) But it's not explicitly the maximum memory consumption either. `add_at` contributes `n_samples * n_clusters` further ints.